dot/core/helpers_test.go

```diff
@@ -17,10 +19,174 @@ import (
 	rtstorage "github.com/ChainSafe/gossamer/lib/runtime/storage"
 	"github.com/ChainSafe/gossamer/lib/runtime/wasmer"
 	"github.com/ChainSafe/gossamer/lib/utils"
+	"github.com/ChainSafe/gossamer/pkg/scale"
 	"github.com/golang/mock/gomock"
 	"github.com/stretchr/testify/require"
 )
 
+func balanceKey(t *testing.T, pub []byte) (bKey []byte) {
+	t.Helper()
+
+	h0, err := common.Twox128Hash([]byte("System"))
+	require.NoError(t, err)
+	bKey = append(bKey, h0...)
+	h1, err := common.Twox128Hash([]byte("Account"))
+	require.NoError(t, err)
+	bKey = append(bKey, h1...)
+	h2, err := common.Blake2b128(pub)
+	require.NoError(t, err)
+	bKey = append(bKey, h2...)
+	bKey = append(bKey, pub...)
+	return
+}
+
+// Note might need to pass in Config but will see
```

Comment:
Please definitely do this. The dark magic checking nil fields in the config and setting in them is just horrible to debug later down the road.

Reply:
Let me know what you think, but instead of passing in I can also just remove all these conditionals I believe. We define an empty config and then do a bunch of nil checks for no reason. So maybe I just can remove all these nil checks for this case? It seems the options are simplify this code and always build up the config (which I feel is prob more straightforward for testing) or be more flexible and take in config as a param to a separate buildConfig method and do all the nil checking there.

I feel like the second might feel nicer but I'm not sure its necessary. 1) it adds complexity since Im not sure in what use case we are passing in a portion config in the core tests and 2) it doesn't remove the nil checks then lol. I would lean towards leaving as is where we construct in this test helper from an empty config and then removing the nil checks to chean it up (and like you said make it easier to debug. I think this makes more sense for this use case, but let me know what you think

Reply:
Either don't pass a config or fully pass it really. In the second case, avoid nil checks so that it force the dev to repeat and be explicit of what's needed.

If we always pass an empty config, then just create it within the helper and remove the nil checks 👍  that's fine